lib/p2p/Peer.ts

```diff
@@ -839,6 +839,12 @@ class Peer extends EventEmitter {
   }
 
   private handleDisconnecting = (packet: packets.DisconnectingPacket): void  => {
+    if (!this.sessionInitPacket && packet.body && packet.body.reason === DisconnectionReason.Banned) {
+      this.recvDisconnectionReason = packet.body.reason;
+      this.emit('reputation', ReputationEvent.ManualBan);
```

Comment:
I don't think we should use the manual ban event here, I think of that for when we are the ones doing the banning. It would prevent the peer from "unbanning" us in the future, since we will have banned them. Rather, we should be making a note that we have been banned by this peer to prevent us from trying to reconnect to them in the future. We could do this with a new database field in the `nodes` table, which we'd toggle in the code below. We'd also need to make sure to toggle it back off if the peer successfully connects with us again in the future.